warehouse/query-core/src/main/java/datawave/query/iterator/QueryIterator.java

```diff
@@ -1328,7 +1329,8 @@ protected IteratorBuildingVisitor createIteratorBuildingVisitor(Class<? extends
                         .setIvaratorCacheBufferSize(this.getIvaratorCacheBufferSize())
                         .setIvaratorCacheScanPersistThreshold(this.getIvaratorCacheScanPersistThreshold())
                         .setIvaratorCacheScanTimeout(this.getIvaratorCacheScanTimeout()).setMaxRangeSplit(this.getMaxIndexRangeSplit())
-                        .setIvaratorMaxOpenFiles(this.getIvaratorMaxOpenFiles()).setIvaratorSources(this, this.getMaxIvaratorSources())
```

Comment:
Does the QueryIterator still need to implement SourceFactory after these changes?

Reply:
Yes, because it is used by the SourceManager which determines how many non-ivarator iterators can be made.